rmf_site_editor/src/widgets/menu_bar.rs

```diff
@@ -160,28 +165,32 @@ impl Plugin for MenuPluginManager {
 
 /// Helper function to render a submenu starting at the entity.
 fn render_sub_menu(
+    state: &State<AppState>,
     ui: &mut Ui,
     entity: &Entity,
     children: &Query<&Children>,
     menus: &Query<(&Menu, Entity)>,
-    menu_items: &Query<(&mut MenuItem, Option<&MenuDisabled>)>,
+    menu_items: &Query<(&mut MenuItem, Has<MenuDisabled>)>,
+    menu_states: &Query<Option<&MenuVisualizationStates>>,
     extension_events: &mut EventWriter<MenuEvent>,
     skip_top_label: bool,
 ) {
+    if let Some(states) = menu_states.get(*entity).ok().flatten() {
+        if !states.contains(state.get()) {
+            return;
+        }
+    }
     if let Ok((e, disabled)) = menu_items.get(*entity) {
         // Draw ui
         match e {
             MenuItem::Text(title) => {
-                if ui
-                    .add_enabled(disabled.is_none(), Button::new(title))
-                    .clicked()
-                {
+                if ui.add_enabled(disabled, Button::new(title)).clicked() {
                     extension_events.send(MenuEvent::MenuClickEvent(*entity));
                 }
             }
             MenuItem::CheckBox(title, mut value) => {
                 if ui
-                    .add_enabled(disabled.is_none(), egui::Checkbox::new(&mut value, title))
+                    .add_enabled(disabled, egui::Checkbox::new(&mut value, title))
```

Comment:
:facepalm: Classic boolean is inverted issue, thanks for giving it a try! Committed separately to avoid DCO issues whenever doing so from the Github web UI https://github.com/open-rmf/rmf_site/pull/189/commits/e3346b5d5b2105777d7f6450a9f8dbe91db0c9d7